pandas/core/internals/managers.py

```diff
@@ -373,23 +373,20 @@ def apply(self: T, f, align_keys=None, **kwargs) -> T:
 
         self._consolidate_inplace()
 
-        align_copy = False
-        if f == "where":
-            align_copy = True
-
         aligned_args = {k: kwargs[k] for k in align_keys}
 
         for b in self.blocks:
 
             if aligned_args:
-                b_items = self.items[b.mgr_locs.indexer]
 
                 for k, obj in aligned_args.items():
                     if isinstance(obj, (ABCSeries, ABCDataFrame)):
-                        axis = obj._info_axis_number
-                        kwargs[k] = obj.reindex(
-                            b_items, axis=axis, copy=align_copy
-                        )._values
+                        # The caller is responsible for ensuring that
+                        #  obj.axes[-1].equals(self.items)
+                        if obj.ndim == 1:
+                            kwargs[k] = obj.iloc[b.mgr_locs.indexer]._values
```

Comment:
you don't like the
```
indexer = tupel([None] * obj.ndim)
indexer[-1] = b.mgr_locs.indexer
kwargs[k] = obj.iloc[indexer]._values
```

?

Reply:
actually i have a note to myself to make iloc handle Ellipsis so this can become `obj.iloc[..., indexer]._values`

xref #10956

Reply:
ok great